src/api/Visualization/types.ts

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-10-30.
+ */
+
+import {FolderChildren, GenericObject, IDataSourceParams, PersistedItem} from '../commonTypes';
+import {VizEdge, VizEdgeInfo, VizNode, VizNodeInfo} from '../graphItemTypes';
+import {IAlternativeIdSettings} from '../DataSource';
+import {GraphQueryDialect} from '../GraphQuery';
+
+export interface IGetVisualizationCountParams extends IDataSourceParams {}
+
+export type GetVisualizationCountResponse = number;
+
+export interface IGetVisualizationParams extends IDataSourceParams {
+  id: number;
+  withDigest?: boolean;
+  withDegree?: boolean;
+}
+
+export enum VisualizationRight {
+  READ = 'read',
+  WRITE = 'write',
+  OWNER = 'owner'
+}
+
+export enum VisualizationMode {
+  NODE_LINK = 'nodelink',
+  GEO = 'geo'
+}
+
+export interface ItemFields {
+  captions: GenericObject<{displayName: boolean; properties: string[]; active: boolean}>;
+  types: GenericObject<'string' | 'number' | 'date-timestamp'>;
+}
+
+export interface VisualizationDesign {
+  // TODO type this
+}
+
+export interface VisualizationGeo {
+  latitudeProperty?: string;
+  longitudeProperty?: string;
+  layers: string[];
+}
+
+export type VisualizationLayout =
```

Comment:
@francescoinfante We have that in front, that's better for us. Can you implement it in rest-client ?

```typescript 
export type VisualizationLayout = ForceAlgorithm | HierarchicalAlgorithm | RadialAlgorithm;

export type ForceLayoutMode = 'best' | 'fast';
export type HierarchicalLayoutMode = 'LR' | 'RL' | 'TB' | 'BT';

export enum Algorithm {
  Force = 'force',
  Hierarchical = 'hierarchical',
  Radial = 'radial',
}

export interface ForceParameters {
  mode: ForceLayoutMode;
}

export interface HierarchicalParameters {
  mode: HierarchicalLayoutMode;
  rootNode?: NodeId;
}

export interface RadialParameters {
  rootNode: NodeId;
}

export interface ForceAlgorithm extends ForceParameters {
  algorithm: Algorithm.Force;
}

export interface HierarchicalAlgorithm extends HierarchicalParameters {
  algorithm: Algorithm.Hierarchical;
}

export interface RadialAlgorithm extends RadialParameters {
  algorithm: Algorithm.Radial;
}
```